subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/LazyAttributesIntegrationTest.groovy

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.attributes
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+
+class LazyAttributesIntegrationTest extends AbstractIntegrationSpec {
+    @ToBeFixedForConfigurationCache
+    def "properties used as attribute values are read lazily"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            Property<String> sampleProperty = project.objects.property(String)
+            sampleProperty.set("original value")
+
+            configurations {
+                sample {
+                    visible = false
+                    canBeResolved = false
+                    canBeConsumed = true
+
+                    attributes {
+                        attribute(Usage.USAGE_ATTRIBUTE, sampleProperty.flatMap(value -> project.provider(() -> objects.named(Usage, value))))
```

Comment:
I think we can do something simpler here...

```
sampleProperty.map(value -> objects.named(Usage, value))
```

Reply:
I think I needed this at one point in the test-fix cycle, but the simpler version works now.